eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1306 @@
+/* Copyright 2018 Endless Mobile, Inc. */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+#include <endless/endless.h>
+#include <glib/gi18n.h>
+#include <json-glib/json-glib.h>
+#include <mustache.h>
+
+#include "eknr-boxed-set-info.h"
+#include "eknr-errors.h"
+#include "eknr-renderer.h"
+
+/**
+ * SECTION:renderer
+ * @title: Renderer
+ * @short_description: A post-processing renderer for HTML article content
+ *
+ * The renderer is responsible for adding some final postprocessing to
+ * articles on the client side before it is displayed to the user. What
+ * postprocessing happens depends on the article's source.
+ */
+struct _EknrRenderer
+{
+  GObject parent_instance;
+};
+
+typedef struct _EknrRendererPrivate
+{
+  GHashTable *cache;
```

Comment:
Nitpick: I like to specify in a comment what the key and value types of `GHashTable`s are.